features/discovery/src/main/java/org/opennms/netmgt/discovery/DiscoveryTaskExecutorImpl.java

```diff
@@ -67,10 +72,16 @@
     @Autowired
     private EventForwarder eventForwarder;
 
+    @Autowired(required = false)
+    private LocationAwareDetectorClient locationAwareDetectorClient;
+
     private final AtomicInteger taskIdTracker = new AtomicInteger();
 
+    private DiscoveryConfigFactory configFactory;
+
     @Override
     public CompletableFuture<Void> handleDiscoveryTask(DiscoveryConfiguration config) {
+        this.configFactory = new DiscoveryConfigFactory(config);
```

Comment:
We should carry this with the DiscoveryJob rather than storing it as a member variable - this breaks thread safety of the `DiscoveryTaskExecutor` implementation.